modules/ding_entity/ding_entity.api.php

```diff
@@ -62,6 +62,55 @@ function hook_ding_entity_buttons($type, $entity, $view_mode, $widget) {
   }
 }
 
+/**
+ * Help determine whether entities belongs to given pseudo-classes.
+ *
+ * Note that there's also a single entity version of this hook named
+ * hook_ding_entity_is(), which is invoked from within a ding entity object and
+ * therefore only support one entity at a time. This hook is therefore better to
+ * implement if acting on multiple entities and the implementation can achieve
+ * better performance by recieving all entities at once.
+ *
+ * @param array $entities
+ *   The entities that needs to be checked against the pseudo-class.
+ * @param string $class
+ *   The pseudo class to check for.
+ *
+ * @return array
+ *   The implementation must return an array with an entry for each of the
+ *   passed entities keyed by their ding_entity_id. If the implementation
+ *   doesn't have a say for a particular entity and/or pseudo-class it must set
+ *   NULL to return something for that entity and signal that it is neutral.
+ */
+function hook_ding_entity_entities_is(array $entities, $class) {
```

Comment:
I understand what you are trying to do here but I am not sure I agree about the approach.

Bundling multiple checks into a single request is nice and efficient but it makes it very hard for developers to know whether they should 1) use 2) implement `ding_entity_entities_is()` or `ding_entity_is()` for their modules.

One approach to this would be to fall back to `ding_entity_is()` for modules and classes which does not support `ding_entity_entities_is()`. This quickly becomes very complex and it may not be possible at all.

Another approach could be to implement a new provider hook á la `fbs_reservation_is_reservable($provider_ids)` where the implementation for FBS would be about the same as in `fbs_ding_entity_entities_is()`.

A third approach might be to use the `reservable` info from the `availability_items` provider hook. If implemented correctly by the provider this could actually provide the correct information. Note that this creates a dependency from ding_reservation against an availability provider making the whole situation a bit more muddy.

Personally I am leaning towards a new provider hook.

I could also be OK with multiple calls to `->is('reservable')` although it may decrease performance.

Whatever you decide on please remember to clean up. #1139 also introduced changes to `connie` and elsewhere which may not be needed anymore.

Reply:
Thanks for the good explanation 👍 I understand your concerns.

I think it's important to get the performance benefit from the bundling. After reading your feedback I also lean towards a new provider hook, so I will implement that instead.

Regarding the changes to connie from #1139: I added the entity_is implementation to ensure to get reservation buttons when testing with that module. I havn't really used this module a lot, so this might be wrong. Would you say that it's not needed?

Reply:
Done